File: experiments/016-LanguageComparisonRepeated/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260917-060211-45590/analysis/own-tests-project/src/main/java/customerimporter/OldZipCode.java
package customerimporter;

import java.util.function.Supplier;

public class OldZipCode extends ZipCode {

    // instance creation

    public static boolean canHandle(String aZipCode) {
        return Character.isDigit(aZipCode.charAt(0));
    }

    // error messages

    public static String invalidOldZipCodeErrorDescription() {
        return "Invalid old zipcode";
    }

    // initialization

    public OldZipCode(String aZipCode) {
        super(aZipCode);
    }

    // accessing

    public Object value() {
        return Integer.parseInt(code);
    }

    // testing

    public boolean isOld() {
        return true;
    }

    public boolean isNew() {
        return false;
    }

    public Object oldZipCodeIfNone(Supplier<Object> aNoneBlock) {
        return value();
    }

    public Object newZipCodeIfNone(Supplier<Object> aNoneBlock) {
        return aNoneBlock.get();
    }

    // validation

    protected void assertIsValid() {
        assertThat(Characters.areAllDigits(code));
        int number = Integer.parseInt(code);
        assertThat(number >= 1000 && number <= 9999);
    }

    protected String invalidZipCodeErrorDescription() {
        return invalidOldZipCodeErrorDescription();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-202259-53403/workdir/src/customerimporter/old_zip_code.py
from customerimporter.zip_code import ZipCode


class OldZipCode(ZipCode):

    # type

    @classmethod
    def can_handle(cls, a_zip_code):
        return a_zip_code[0].isdigit()

    # error messages

    @classmethod
    def invalid_value_error_description(cls):
        return "Invalid old zipcode"

    # validation

    @classmethod
    def assert_is_valid(cls, a_zip_code):
        cls.assert_is_valid_zip_code_number(a_zip_code)

    # testing

    def is_old(self):
        return True

    # zip code

    def old_zip_code_if_none(self, a_none_block):
        return self.value()

    def value(self):
        return int(self._value)
